src/server/scripts/Spells/spell_monk.cpp

```diff
@@ -252,6 +254,30 @@ class spell_monk_provoke : public SpellScript
     }
 };
 
+// 107428 - Rising Sun Kick
+class spell_monk_rising_sun_kick : public SpellScript
+{
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_MONK_COMBAT_CONDITIONING, SPELL_MONK_MORTAL_WOUNDS });
+    }
+
+    void HandleOnHit(SpellEffIndex /*effIndex*/)
+    {
+        Unit* caster = GetCaster();
+
+        if (caster->HasAura(SPELL_MONK_COMBAT_CONDITIONING))
+            caster->CastSpell(GetHitUnit(), SPELL_MONK_MORTAL_WOUNDS, CastSpellExtraArgs()
+                .SetTriggeringSpell(GetSpell())
+                .SetTriggerFlags(TRIGGERED_IGNORE_CAST_IN_PROGRESS | TRIGGERED_DONT_REPORT_CAST_ERROR));
+    }
+
+    void Register() override
+    {
+        OnEffectHitTarget += SpellEffectFn(spell_monk_rising_sun_kick::HandleOnHit, EFFECT_0, SPELL_EFFECT_TRIGGER_SPELL);
```

Comment:
should it hit every target or just the primary target?

Reply:
it has a 10 seconds recovery time, so you cant hit two targets at the same time

Reply:
cooldown doesnt matter (windwalker reduces it by pressing blackout kick) - aura is not unique, all targets hit get it